core/src/test/java/com/es/core/cart/HttpSessionCartServiceUnitTest.java

```diff
@@ -0,0 +1,93 @@
+package com.es.core.cart;
+
+import com.es.core.model.phone.CartItem;
+import com.es.core.model.phone.Phone;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.Spy;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.math.BigDecimal;
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class HttpSessionCartServiceUnitTest {
+    @InjectMocks
+    @Spy
+    private HttpSessionCartService cartService;
+
+    @Mock
+    private Cart cart;
+
+    @Test
+    public void getCartTotalPriceTest() {
+        double[] prices = {200, 350, 400.5f, 150};
```

Comment:
Don't do that in tests. Ideally, all mocks must be defined as `@Mock` fields of the test class (phone1, phone2, ... or whatever) and set up in the `@Before` method according to the happy path or more or less the common state. The test method should normally only consist of 2 or 3 extremely short parts:
1) (optional) Change the default mock setup defined in the `@Before` method, to test some boundary condition (not-happy-path tests)
2) Invoke the tested method. If applicable, save its result to a variable
3) Verify the result using assert* / verify methods

Reply:
So, in fact, after all changes this method should look like:
```java
when(cart.getCartItems()).thenReturn(Arrays.asList(phone1, phone2, phone3, phone4)); // if not set up in @Before, which better to be, actually

BigDecimal result = cartService.getTotalPrice();

assertEquals(TOTAL_PRICE, result); // where TOTAL_PRICE is a class constant
```

Reply:
fixed